File: flink/flink_kafka_scala_demo_1/src/main/scala/demo/flink/scala/one/util/FlinkUtil.scala
package demo.flink.scala.one.util


import demo.flink.scala.one.config.AppConfig
import org.apache.flink.api.common.restartstrategy.RestartStrategies
import org.apache.flink.streaming.api.TimeCharacteristic
import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment

class FlinkUtil(config: AppConfig) {

  def getExecutionContext(): StreamExecutionEnvironment = {
    val env = StreamExecutionEnvironment.getExecutionEnvironment
    env.getConfig.setUseSnapshotCompression(config.getBoolean("task.checkpointing.compressed", false))
    env.enableCheckpointing(config.getConfig().getInt("task.checkpointing.interval"))
    env.getCheckpointConfig.setCheckpointTimeout(config.getConfig().getLong("task.checkpointing.timeout"))
    env.setStreamTimeCharacteristic(TimeCharacteristic.IngestionTime)
    env.setRestartStrategy(RestartStrategies.fixedDelayRestart(config.getConfig().getInt("task.restart-strategy.attempts"), config.getConfig().getLong("task.restart-strategy.delay")))
    env
  }
}

File: flink/flink_kafka_java_demo_1/src/main/java/demo/flink/util/FlinkUtil.java
package demo.flink.util;

import com.typesafe.config.Config;
import org.apache.flink.api.common.restartstrategy.RestartStrategies;
import org.apache.flink.streaming.api.TimeCharacteristic;
import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;

public class FlinkUtil {
    public  StreamExecutionEnvironment getExecutionContext(Config config) {
        StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
        env.getConfig().setUseSnapshotCompression(config.getBoolean("task.checkpointing.compressed"));
        env.enableCheckpointing(config.getInt("task.checkpointing.interval"));
        env.getCheckpointConfig().setCheckpointTimeout(config.getLong("task.checkpointing.timeout"));
        env.setStreamTimeCharacteristic(TimeCharacteristic.IngestionTime);
        env.setRestartStrategy(RestartStrategies.fixedDelayRestart(config.getInt("task.restart-strategy.attempts"), config.getLong("task.restart-strategy.delay")));
        return env;
    }
}
